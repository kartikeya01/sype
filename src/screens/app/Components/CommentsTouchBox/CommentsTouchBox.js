import React from 'react';
import {Caption} from 'react-native-paper';
import {StyleSheet, TouchableWithoutFeedback} from 'react-native';
import AirbnbPropTypes from 'airbnb-prop-types';

export default class CommentsTouchBox extends React.Component {
  constructor(props) {
    super(props);
  }

  handleCommentClick = () => {};

  render() {
    return (
      <TouchableWithoutFeedback
        onPress={() => console.log('go to view comments')}>
        <Caption style={styles.captionStyle}>
          View All {this.props.commentsLength} Comment
          {this.props.commentsLength <= 1 ? '' : 's'}
        </Caption>
      </TouchableWithoutFeedback>
    );
  }
}
const styles = StyleSheet.create({
  captionStyle: {
    fontSize: 14,
  },
});

CommentsTouchBox.propTypes = {
  commentsLength: AirbnbPropTypes.nonNegativeInteger.isRequired,
};
