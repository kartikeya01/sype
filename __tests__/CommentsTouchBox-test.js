import 'react-native';
import React from 'react';
import CommentsTouchBox from '@screens/app/Components/CommentsTouchBox/CommentsTouchBox';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
var assert = require('assert');

test('like button snapshot test', () => {
  const treeWithoutCommentsLength = renderer
    .create(<CommentsTouchBox />)
    .toJSON();
  expect(treeWithoutCommentsLength).toMatchSnapshot();
  const treeWithCommentLengthZero = renderer
    .create(<CommentsTouchBox commentsLength={0} />)
    .toJSON();
  expect(treeWithCommentLengthZero).toMatchSnapshot();
  const treeWithCommentLengthOne = renderer
    .create(<CommentsTouchBox commentsLength={1} />)
    .toJSON();
  expect(treeWithCommentLengthOne).toMatchSnapshot();
  const treeWithCommentLengthGreaterThanOne = renderer
    .create(<CommentsTouchBox commentsLength={2} />)
    .toJSON();
  expect(treeWithCommentLengthGreaterThanOne).toMatchSnapshot();
});

function testPropTypesWithoutErrors(commentsLength) {
  it(
    'testing commentsLength values (No errors) where commentsLength is ' +
      commentsLength,
    () => {
      const wrapper = shallow(
        <CommentsTouchBox commentsLength={commentsLength} />,
      );
      const instance = wrapper.instance();
      const commentsLengthProp = instance.props.commentsLength;
      expect(commentsLength).toEqual(commentsLengthProp);
    },
  );
}

function testPropTypesWithErrors(commentLength, errorMsg, testDescription) {
  it('Throws failed propType Error for ' + testDescription, () => {
    let result = checkPropTypes(
      CommentsTouchBox.propTypes,
      {commentsLength: commentLength},
      'prop',
      CommentsTouchBox.commentsLength,
    );
    assert(result === errorMsg);
  });
}

describe('testing commentsLength values', () => {
  testPropTypesWithoutErrors(0);
  testPropTypesWithoutErrors(1);
  testPropTypesWithoutErrors(1000);
  testPropTypesWithErrors(
    undefined,
    'Failed prop type: commentsLength in undefined must be an integer',
    'undefined value given to commentsLength',
  );
  testPropTypesWithErrors(
    null,
    'Failed prop type: commentsLength in undefined must be an integer',
    'null value given to commentsLength',
  );
  testPropTypesWithErrors(
    '',
    'Failed prop type: commentsLength in undefined must be an integer',
    'empty string value given to commentsLength',
  );
  testPropTypesWithErrors(
    'test',
    'Failed prop type: commentsLength in undefined must be an integer',
    'string value with length > 0, given to commentsLength',
  );
  testPropTypesWithErrors(
    true,
    'Failed prop type: commentsLength in undefined must be an integer',
    'true bool value given to commentsLength',
  );
  testPropTypesWithErrors(
    false,
    'Failed prop type: commentsLength in undefined must be an integer',
    'false bool value given to commentsLength',
  );
  testPropTypesWithErrors(
    234.5,
    'Failed prop type: commentsLength in undefined must be an integer',
    'positive float given to commentsLength',
  );
  testPropTypesWithErrors(
    -1.5,
    'Failed prop type: commentsLength in undefined must be an integer',
    'negative float given to commentsLength',
  );
  testPropTypesWithErrors(
    -1,
    'Failed prop type: commentsLength in undefined must be a non-negative number',
    'negative integer given to commentsLength',
  );
});
