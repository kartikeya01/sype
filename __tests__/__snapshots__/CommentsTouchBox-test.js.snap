// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`like button snapshot test 1`] = `
<Text
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
      ],
    ]
  }
>
  View All 
   Comment
  s
</Text>
`;

exports[`like button snapshot test 2`] = `
<Text
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
      ],
    ]
  }
>
  View All 
  0
   Comment
  
</Text>
`;

exports[`like button snapshot test 3`] = `
<Text
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
      ],
    ]
  }
>
  View All 
  1
   Comment
  
</Text>
`;

exports[`like button snapshot test 4`] = `
<Text
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "fontSize": 14,
          },
        ],
      ],
    ]
  }
>
  View All 
  2
   Comment
  s
</Text>
`;
